### PR TITLE
Improvements to the End of rains Dilogue

### LINKED ISSUE #5196: Improvements to End of Rains/Season dialog layout
I'm sure this is in another issue but I can't find it easily and good to have it separate here.
Once my pull request #5195 on the calculation is merged the dialog can also be improved:

1. Checkboxes for End of Rains/Season change to radio buttons at the top of the dialog
2. There is then much more space on the dialog for the controls as the End of Rains controls can be on top of the End of Season controls now that they won't display together. In particular the text boxes for the new column names are very small and could be a lot bigger now. Consider the size and spacing of other controls.
3. More descriptive labels. For End of Rains currently it just says "Amount" and "Over day(s)". I think the dialogs should be more descriptive and give you a good idea of what the calculation might do just from reading it. 
So this could change to "End of rains when > amount:" for example. It would be good for someone to look critically at all the labels and consider if they are clear. This is good for someone who doesn't know the climatic calculations well. If you can't work out what the general idea of what the calculation will do, then its probably not clear enough.

@africanmathsinitiative/developers this would be a good task for someone interested in getting involved in the climatic menu in general. Comment if you are interested - it could be more than one person.

### PR DESCRIPTION
**Pull request by Ivanluv**
Fixes #5196 @africanmathsinitiative/developers  this is ready for review

### CODE REVIEW

**Comment by Muthenya** [comment]
Reviewing this.

File: instat/dlgEndOfRainsSeason.vb [hunk 1]
```diff
@@ -1049,17 +1053,20 @@ Public Class dlgEndOfRainsSeason
...
             clsRunCalculation.AddParameter("calc", clsRFunctionParameter:=clsEndRainsCombinationCalc)
-        ElseIf ucrChkEndOfSeason.Checked Then
+        ElseIf rdoEndOfSeasons.Checked Then
             clsRunCalculation.AddParameter("calc", clsRFunctionParameter:=clsEndSeasonCombinationCalc)
-        Else
-            clsRunCalculation.RemoveParameterByName("calc")
         End If
     End Sub
+
+    Private Sub ucrPnlEndOfRainsAndSeasons_Load(sender As Object, e As EventArgs) Handles ucrPnlEndOfRainsAndSeasons.Load
```
Comment (Muthenya) [inline instat/dlgEndOfRainsSeason.vb]: This is an empty sub. Kindly remove it.

File: instat/dlgEndOfRainsSeason.vb [hunk 2]
```diff
@@ -883,9 +887,9 @@ Public Class dlgEndOfRainsSeason
         Dim bOkEnabled As Boolean = True
 
         'TODO change to radio buttons so only one can be checked at a time
-        If ucrChkEndOfRains.Checked AndAlso ucrChkEndOfSeason.Checked Then
-            bOkEnabled = False
-        ElseIf ucrChkEndOfRains.Checked Then
+        'If ucrChkEndOfRains.Checked AndAlso ucrChkEndOfSeason.Checked Then
+        'bOkEnabled = False
```
Comment (Muthenya) [inline instat/dlgEndOfRainsSeason.vb]: These two lines can now be removed.

File: instat/dlgEndOfRainsSeason.vb [hunk 3]
```diff
@@ -870,10 +874,10 @@ Public Class dlgEndOfRainsSeason
...
         'TODO temporary until these are converted to radio buttons
         If bReset Then
-            ucrChkEndOfRains.Checked = True
-            ucrChkEndOfRains.Checked = False
-            ucrChkEndOfSeason.Checked = True
-            ucrChkEndOfSeason.Checked = False
+            'ucrChkEndOfRains.Checked = True
+            'ucrChkEndOfRains.Checked = False
+            'ucrChkEndOfSeason.Checked = True
+            ' ucrChkEndOfSeason.Checked = False
```
Comment (Muthenya) [inline instat/dlgEndOfRainsSeason.vb]: These can now be removed.

File: instat/dlgEndOfRainsSeason.vb [hunk 4]
```diff
@@ -848,7 +852,7 @@ Public Class dlgEndOfRainsSeason
 
         ucrReceiverRainfall.SetRCode(clsRollSumRainFunction, bReset)
 
-        ucrChkEndOfSeason.SetRCode(clsEndRainsCombinationSubCalcList, bReset)
+        'ucrChkEndOfSeason.SetRCode(clsEndRainsCombinationSubCalcList, bReset)
```
Comment (Muthenya) [inline instat/dlgEndOfRainsSeason.vb]: This should be removed.

File: instat/dlgEndOfRainsSeason.vb [hunk 5]
```diff
@@ -205,10 +205,12 @@ Public Class dlgEndOfRainsSeason
...
 #Region "end_of_rains_controls"
         'TODO Set conditions for end of rains vs end of season
-        ucrChkEndOfRains.SetText("End of Rains")
-        ucrChkEndOfRains.AddToLinkedControls(ucrNudAmount, {True}, bNewLinkedHideIfParameterMissing:=True)
-        ucrChkEndOfRains.AddToLinkedControls(ucrNudTotalOverDays, {True}, bNewLinkedHideIfParameterMissing:=True)
-        ucrChkEndOfRains.AddToLinkedControls(ucrChkEndofRainsDoy, {True}, bNewLinkedHideIfParameterMissing:=True)
+
+        ucrPnlEndOfRainsAndSeasons.AddToLinkedControls({ucrNudAmount, ucrNudTotalOverDays}, {rdoEndOfRains}, bNewLinkedHideIfParameterMissing:=True)
+        ucrPnlEndOfRainsAndSeasons.SetLinkedDisplayControl(grpEndofRains)
+        'rdoEndOfRains.Checked
```
Comment (Muthenya) [inline instat/dlgEndOfRainsSeason.vb]: This should be remove along with the spaces that follow it.

**Comment by Muthenya** [comment]
Just a few more issues arising with the dialog.

1. When you load the dialog for the first time, there are some issues with with the designer as shown below. I think this is to do with the linking of the controls to their respective radio buttons. This should be done controls on all the required events.

Capture5

2. The reset is not set correctly as well.

**Comment by stevenndungu** [comment]
Reviewing this @Ivanluv

**Comment by stevenndungu** [comment]
@Ivanluv  this looks great only that you have not added in option 3- adding more descriptive labels.

**Comment by stevenndungu** [comment]
I think you can do that quickly

**Comment by maxwellfundi** [comment]
@Ivanluv  I thought of this during the workshop and good that it has made some progress, what issues are remaining? Any help needed?

**Comment by Ivanluv** [comment]
adding more descriptive names to some controls.I am not sure on how to name them so I request you help with this

**Comment by dannyparsons** [comment]
I'm happy to leave the name changes for another issue, as that will need input from others including Roger.

Is this ok apart from this, from the reviewers?

**Comment by shadrackkibet** [comment]
@Ivanluv what is the status of this?

**Comment by Ivanluv** [comment]
@shadrackkibet  its ready for review

**Comment by shadrackkibet** [comment]
Testing this now.

**Comment by shadrackkibet** [comment]
This is working well no major bug at least for now, only a few minor issues to sort out.

- Sort out tab order.
- When I am on End of seasons and I just check day then close and reopen the dialog I find occurence checkbox checked.
- On End of rains, when I just check day then click ok and then reopen the dialog, I find date and occurrence textboxes is cleared.
- When I first open the dialog then check End of seasons radiobutton then close and reopen, check end rains radiobutton I find the amount nud resets to some low negative value. See below.
image

**Comment by rdstern** [comment]
And a trivial change - could the radio button be End of Season rather than End Of Seasons.  Thanks

**Comment by shadrackkibet** [comment]
We have now fixed issues with this. @dannyparsons this is ready for you to review.

**Comment by dannyparsons** [comment]
@rdstern are you able to test this?

**Comment by shadrackkibet** [comment]
@rdstern?

**Comment by rdstern** [comment]
This looks much better.  Great.
So far my only comment is on the end of the season.  I would prefer the controls for the Evaporation to be above those for saving the results.  This was the previous version.  I don't know if there was a reason for the change in order? 

image

I still have to test that the results are ok, but I don't think the code for the calculations has changed?  Is that true?  If so, then I would be happy for the new version to be merged - hopefully after the small change in the order of the controls.

**Comment by rdstern** [comment]
No there is one more change, that might need a more general discussion.  The dialogue is quite "long".  In a recent workshop I used a setting of 800 by 600 for the screen and then couldn't see the bottom controls on some of the dialogues.  This dialogue seems to long and could now perhaps be shortened.  This is particularly as now the rains and season are separate - so it can easily be shorter than before.

**Comment by shadrackkibet** [comment]
The size can be adjusted accordingly, especially on the end of rains there is too much space below the save controls.

**Comment by dannyparsons** [comment]
@Ivanluv should this be tested again?

**Comment by Ivanluv** [comment]
yes it can be tested

**Comment by dannyparsons** [comment]
@shadrackkibet @rdstern are you able to check this again?

**Comment by shadrackkibet** [comment]
Yes, testing this again.

**Comment by shadrackkibet** [comment]
Generally, the dialog works fine. I still feel that the design can be enhanced. There is too much space above save controls on both end of rains and end season. This should be pushed up and adjust the size of the dialog accordingly to accommodate other controls when on different options.  Have a look at how this has been done in the calculator where height, width and position of controls are adjusted automatically by a multiplier. This will help to reduce the height of the dialog and hence solve @rdstern concern above. When on End season the save controls are not properly centred in the group box. See below.
image
image

**Comment by rdstern** [comment]
It looks ready to have a small adjustment to the dialogue and then be merged.  The main thing I wanted is now done, i.e. to move the Evaporation controls up.
There is space for some small adjustments in the positioning of the controls, but not as much as indicated above.  Here is the dialogue with the end of season and variable evaporation:
image

a) Please move the Day, Date and Occurrence contols down a bit - same as with the End of rains.
b) The group box can be made a bit smaller
c) There can be a bit less space between Capacity and Evaporation.
d) Hence the dialogue can be a bit shorter.  That will do for now.

Then hopefully it can be merged.  We should probably see the layout one last time.

**Comment by Ivanluv** [comment]
is this okay?
image

**Comment by rdstern** [comment]
Better.  But there is still too little space between "End of Season" and the line below.  And there is quite a lot of space above the End of Season group-box.  So move it up a bit and add the same space as you have between End of Rains and the line below there.

**Comment by Ivanluv** [comment]
what about this?
image

**Comment by rdstern** [comment]
Great.  Let's see if @dannyparsons is now happy to merge.

**Comment by dannyparsons** [comment]
The design looks good. What is the height of the dialog? We do need to take seriously the issue of size on low resolution screens. Targeting 800x600 means the height should be less than 600. Tab order and such can help but I think we need to enforce a max size.

**Comment by Ivanluv** [comment]
Its heigt is 643

**Comment by maxwellfundi** [comment]
@dannyparsons  What are your thoughts about this now?

File: instat/dlgEndOfRainsSeason.vb [hunk 6]
```diff
@@ -460,6 +461,8 @@ Public Class dlgEndOfRainsSeason
 
         ucrSelectorForWaterBalance.Reset()
         ucrReceiverRainfall.SetMeAsReceiver()
+        'temporary fix
+        rdoEndOfRains.Checked = True
```
Comment (dannyparsons) [inline instat/dlgEndOfRainsSeason.vb]: Is there a problem with the conditions for this to be needed?

**Comment by rdstern** [comment]
I have had a quick check and all looks fine (so far).  I checked for Dodoma - so just single station, but both end of rains and end of season.  With simple and also varying evaporation.
I like the new design.  I suggest it be merged and I'll continue with a few more checks then.

File: instat/dlgEndOfRainsSeason.vb (hunk 6 above)
Comment (Ivanluv) [reply instat/dlgEndOfRainsSeason.vb]: Yes,all the radio buttons have almost similar functions and parameters ,I was stuck on how to  implement the conditions

**Pull request merged**